EIPS/eip-draft_grant_standard.md

```diff
@@ -26,69 +26,282 @@ The current process for grants in the Ethereum ecosystem is for membership organ
-       * if run when `minimum_token_threshold` < `current_token_signal`
-           * tallies up the votes and/or checks money according to `type`'s rules
-           * calls `payout`
-  * for `VOTE_THRESHOLD`:
-       * tallies up the votes
-       * calls `payout` if threshold is reached
-  * for `FUND_THRESHOLD`:
-       * tallies up the funds
-       * calls `payout` if threshold is reached
-  * for `OPAQUE`:
-       * shrugs. Run the function.
-
-TODO:
-* Interface definition with Solidity
-* Reasoning on what to leave out of the standard vs what to bake in (what do we want to be immutable?)
+
+
+NOTE: We use the term `GRAINS` to refer to the smallest denomination of an ERC20 `currency` similar to how `WEI` is used to denote the smallest denomination of Ether.
+
+
+
+### Types
+
+#### Grantee
+```
+struct Grantee {
+    uint256 targetFunding;   // Funding amount targeted for Grantee.
+    uint256 totalPayed;      // Cumulative funding received by Grantee.
+    uint256 payoutApproved;  // Pending payout approved by Manager.
+}
+```
+
+#### Donor
+```
+struct Donor {
+    uint256 funded;          // Total amount funded.
+    uint256 refunded;        // Cumulative amount refunded.
+}
+```
+
+### Getter Methods
+
+#### manager
+Returns the multisig, EOA, or other address responsible for managing the grant. 
+```
+function manager() public view returns(address)
+```
+
+#### currency
+Returns the grant currency. If null, amount is in wei, otherwise address of ERC20-compliant (or other currency) contract.
+```
+function currency() public view returns(address)
+```
+
+#### targetFunding
+Returns the grant funding threshold required to begin releasing funds.
+```
+function targetFunding() public view returns(uint256)
+```
+
+#### fundingExpiration
+Returns the date after which funding must be complete.
+```
+function fundingExpiration() public view returns(uint256)
+```
+
+#### contractExpiration
+Returns the date after which payouts must be complete or anyone can trigger refunds.
+```
+function contractExpiration() public view returns(uint256)
+```
+
+#### grantCancelled
+Returns a flag to indicate whether or not grant is cancelled.
+```
+function grantCancelled() public view returns(bool)
+```
+
+#### grantees
+Returns Grantee.
+```
+function grantees(address) public view returns(Grantee)
+```
+
+#### donors
+Returns Donor.
+```
+function donors(address) public view returns(Donor)
+```
+
+#### getAvailableBalance
+Get available grant balance.
+```
+function getAvailableBalance() public view returns(uint256);
+```
+
+#### totalFunding
+Cumulative funding donated by donors.
+```
+function totalFunding() public view returns(uint256);
+```
+
+#### totalPayed
+Cumulative funding payed to grantees.
+```
+function totalPayed() public view returns(uint256);
+```
+
+#### totalRefunded
+OPTIONAL: If refunds permitted, cumulative funding refunded to donors.
```

Comment:
```suggestion
OPTIONAL: If refunds are permitted, this returns the cumulative funding refunded to donors.
```
Also: what's the reasoning behind making this optional? I'd make it return 0 by default if refunds aren't permitted.